pkg/virtualenv/shellgetter/macos_login_shell_cmd_getter.go

```diff
@@ -0,0 +1,60 @@
+package shellgetter
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type macOsLoginShellCmdGetter struct {
+	cmdGetter MacOsUserShellCmdGetter
+}
+
+func newMacOsLoginShellCmdGetter(cmdGetter MacOsUserShellCmdGetter) *macOsLoginShellCmdGetter {
+	return &macOsLoginShellCmdGetter{
+		cmdGetter: cmdGetter,
+	}
+}
+
+// Get detects which shell to run by getting the user's default login shell from /etc/passwd.
+func (g *macOsLoginShellCmdGetter) Get() (string, error) {
+	line, err := g.cmdGetter.RunDsclUserShell()
+	if err != nil {
+		return "", fmt.Errorf("could not get current user's login shell: %w", err)
+	}
+
+	shellCmd := g.parseShellCmd(line)
+
+	return shellCmd, nil
+}
+
+func (*macOsLoginShellCmdGetter) parseShellCmd(line string) string {
+	split := strings.Split(line, ":")
+	afterColon := split[len(split)-1]
+	return strings.TrimSpace(afterColon)
+}
+
+// MacOsUserShellCmdGetter gets the user's login shell on macOS
+type MacOsUserShellCmdGetter interface {
```

Comment:
This is a wrapper for the exec.Command thing on line 53. We need this to be able to unit test the whole thing. Adds lots of extra code, though :\